### PR TITLE
Clarified when methods should end in Async

### PR DESCRIPTION
**Pull request by rpetrusha**
## Clarified when methods should end in Async

Related to #10734 
Addresses dotnet/roslyn#33082

### CODE REVIEW

File: docs/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap.md [hunk 1]
```diff
@@ -15,21 +15,22 @@ ms.assetid: 8cef1fcf-6f9f-417c-b21f-3fd8bac75007
...
...his is in contrast to the Asynchronous Programming Model (APM or `IAsyncResult`) pattern, which requires `Begin` and `End` methods, and in contrast to the Event-based Asynchronous Pattern (EAP), which requires a method that has the `Async` suffix and also requires one or more events, event handler delegate types, and `EventArg`-derived types. Asynchronous methods in TAP include the `Async` suffix after the operation name for methods that return awaitable types, such as <xref:System.Threading.Tasks.Task>, <xref:System.Threading.Tasks.Task%601>, <xref:System.Threading.Tasks.ValueTask>, and <xref:System.Threading.Tasks.ValueTask%601>. For example, an asynchronous `Get` operation that returns a `Task<String>` can be named `GetAsync`. If you're adding a TAP method to a class that already contains an EAP method name with the `Async` suffix, use the suffix `TaskAsync` instead. For example, if the class already has a `GetAsync` method, use the name `GetTaskAsync`. If a method starts an asynchronous operation but does not return an awaitable type, its name should start with `Begin`, `Start`, or some other verb to suggest that this method does not return or throw the result of the operation.
```
Comment (kendrahavens) [inline docs/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap.md]: Looks good. Maybe break up the second sentence:
TAP uses a single method to represent the initiation and completion of an asynchronous operation. This contrasts both the Asynchronous Programming Model (APM or `IAsyncResult`) pattern and the Event-based Asynchronous Pattern (EAP). APM requires `Begin` and `End` methods. EAP requires a method that has the `Async` suffix and also requires one or more events, event handler delegate types, and `EventArg`-derived types.

**Review by kendrahavens — APPROVED** [review]

**Comment by kendrahavens** [comment]
@AArnott may be good to review.

File: docs/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap.md [hunk 2]
```diff
@@ -15,21 +15,22 @@ ms.assetid: 8cef1fcf-6f9f-417c-b21f-3fd8bac75007
...
   
- The parameters of a TAP method should match the parameters of its synchronous counterpart, and should be provided in the same order.  However, `out` and `ref` parameters are exempt from this rule and should be avoided entirely. Any data that would have been returned through an `out` or `ref` parameter should instead be returned as part of the `TResult` returned by <xref:System.Threading.Tasks.Task%601>, and should use a tuple or a custom data structure to accommodate multiple values. 
+ The parameters of a TAP method should match the parameters of its synchronous counterpart and should be provided in the same order.  However, `out` and `ref` parameters are exempt from this rule and should be avoided entirely. Any data that would have been returned through an `out` or `ref` parameter should instead be returned as part of the `TResult` returned by <xref:System.Threading.Tasks.Task%601>, and should use a tuple or a custom data structure to accommodate multiple values.
```
Comment (AArnott) [inline docs/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap.md]: > a TAP method should match the parameters of its synchronous counterpart

It feels like we should encourage folks to add a `CancellationToken` parameter to the new async method even if the synchronous method doesn't offer one.

**Review by AArnott — APPROVED** [review]

File: docs/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap.md (hunk 1 above)
Comment (rpetrusha) [reply docs/standard/asynchronous-programming-patterns/task-based-asynchronous-pattern-tap.md]: Thanks, @kendrahavens. That breaks down a long sentence nicely.

**Review by BillWagner — APPROVED** [review]
This LGTM @rpetrusha 

You can :shipit: when ready.

**Pull request merged**